src/content/app/tools/blast/views/blast-submission-results/BlastSubmissionResults.tsx

```diff
@@ -33,7 +46,117 @@ const BlastSubmissionResults = () => {
 };
 
 const Main = () => {
-  return <div>Here will be submission details</div>;
+  const { submissionId } = useParams() as { submissionId: string };
+  const blastSubmission = useAppSelector((state) =>
+    getBlastSubmissionById(state, submissionId)
+  );
+
+  if (!blastSubmission) {
+    return null;
+  }
+
+  const { submittedData, results } = blastSubmission;
+  const resultsGroupedBySequence = submittedData.sequences.map((sequence) => {
+    const blastResults = results.filter((r) => r.sequenceId === sequence.id);
+    return {
+      sequence,
+      species: submittedData.species,
+      blastResults
+    };
+  });
+
+  const sequenceBoxes = resultsGroupedBySequence.map((data) => (
+    <SequenceBox
+      key={data.sequence.id}
+      species={data.species}
+      sequence={data.sequence}
+      blastResults={data.blastResults}
+    />
+  ));
+
+  return (
+    <div className={styles.blastSubmissionResultsContainer}>
+      <BlastSubmissionHeader
+        submission={blastSubmission}
+        isAnyJobRunning={false}
+      />
+      {sequenceBoxes}
+    </div>
+  );
+};
+
+type SequenceBoxProps = {
+  sequence: {
+    id: number;
+    value: string;
+  };
+  species: Species[];
+  blastResults: BlastResult[];
+};
+
+const SequenceBox = (props: SequenceBoxProps) => {
+  const { sequence, species, blastResults } = props;
+
+  return (
+    <>
```

Comment:
Don't think there's a need to use fragments here since the `div` below is the parent for all the other elements in this component.

Reply:
Also, no need to use the key prop on that div. Keys are required in an array of React elements; but not for a single React element.